pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -493,15 +494,22 @@ func deployBackupBucketInGarden(ctx context.Context, k8sGardenClient client.Clie
 
 	ownerRef := metav1.NewControllerRef(seed, gardencorev1beta1.SchemeGroupVersion.WithKind("Seed"))
 
-	_, err := controllerutils.CreateOrGetAndStrategicMergePatch(ctx, k8sGardenClient, backupBucket, func() error {
+	_, err := controllerutils.CreateOrGetAndMergePatch(ctx, k8sGardenClient, backupBucket, func() error {
 		metav1.SetMetaDataAnnotation(&backupBucket.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
 		backupBucket.OwnerReferences = []metav1.OwnerReference{*ownerRef}
 		backupBucket.Spec = gardencorev1beta1.BackupBucketSpec{
 			Provider: gardencorev1beta1.BackupBucketProvider{
 				Type:   seed.Spec.Backup.Provider,
 				Region: region,
 			},
-			ProviderConfig: seed.Spec.Backup.ProviderConfig,
+			ProviderConfig: &runtime.RawExtension{
+				Raw: func() []byte {
+					if seed.Spec.Backup.ProviderConfig != nil {
+						return seed.Spec.Backup.ProviderConfig.Raw
+					}
+					return nil
+				}(),
+			},
```

Comment:
This looks odd. Why is this change needed? I don't think we do it this way for similar places, do we?

Reply:
I agree with your point. The change was indeed not necessary. I have reverted the `ProviderConfig` setting back to its original form.